l2geth/miner/worker.go

```diff
@@ -929,23 +929,6 @@ func (w *worker) commitNewTx(tx *types.Transaction) error {
 	parent := w.chain.CurrentBlock()
 	num := parent.Number()
 
-	// Preserve liveliness as best as possible. Must panic on L1 to L2
-	// transactions as the timestamp cannot be malleated
-	if parent.Time() > tx.L1Timestamp() {
-		log.Error("Monotonicity violation", "index", num)
-		if tx.QueueOrigin() == types.QueueOriginSequencer {
-			tx.SetL1Timestamp(parent.Time())
-			prev := parent.Transactions()
-			if len(prev) == 1 {
-				tx.SetL1BlockNumber(prev[0].L1BlockNumber().Uint64())
-			} else {
-				log.Error("Cannot recover L1 Blocknumber")
-			}
-		} else {
-			log.Error("Cannot recover from monotonicity violation")
-		}
-	}
-
```

Comment:
note that we did not apply the latest changes from upstream (and I haven't checked what is there) so perhaps deleting this might break other components we have not updated yet.

Reply:
so perhaps we should first update to the tip of master and then figure out compatibility? 

things like this: https://github.com/ethereum-optimism/optimism/commit/2e7f6a55ef1b58f668b7a5ca423f08d5497a7836